contracts/V1/StateChannelDiamondProxy/DisputeManagerFacet.sol

```diff
@@ -99,121 +99,168 @@ contract DisputeManagerFacet is StateChannelCommon {
                 .executeStateTransitionOnState(channelId, encodedState, _tx);
     }
 
-    function getDispute(
-        bytes32 channelId
-    ) public view returns (Dispute memory) {
-        return disputes[channelId];
+    
+
+    function _handleBlockFraudProofs(
+       
+    ) internal {
+      
+       
+
+    }
+
+    function _handleDisputeFraudProofs(
+        // core specific checks
+        // apply proofs
+        // apply dispute to latest state
+        // set state
+    ) internal {
+
+    }
+
+    function _handleTimeoutDispute(
+
+    ) internal {
+
     }
 
     function createDispute(
         bytes32 channelId,
         uint forkCnt,
-        bytes memory encodedLatestFinalizedState,
-        bytes memory encodedLatestCorrectState,
-        ConfirmedBlock[] memory virtualVotingBlocks,
-        address timedoutParticipant,
-        uint foldedTransactionCnt,
-        Proof[] memory proofs
-    ) public {
-        require(!isDisputeInProgress(channelId), ErrorDisputeInProgrees());
-        require(getForkCnt(channelId) == forkCnt, ErrorDisputeForkMismatch());
+        bytes calldata encodedLatestState,
+        StateProof[] calldata stateProofs,
+        address[] calldata onChainSlashes,
+        Proof[] calldata proofs
+    ) public { 
+        // sanity checks
+        if(onChainSlashes.length != onChainSlashedParticipants.length 
+        || onChainSlashes[onChainSlashes.length - 1] != onChainSlashedParticipants[onChainSlashedParticipants.length - 1]) {
```

Comment:
this is not correct in general and would fail when when length == 0 since you're trying to access at position -1, either loop through and compare or hash and compare hashes